### PR TITLE
RHOAIENG-5150: Initial elastic odh application addition

### PR DESCRIPTION
**Pull request by dgutride**
## Description

Adds new manifest for elastic search - vector db solution based on collaboration with them.
Closes: https://issues.redhat.com/browse/RHOAIENG-5150

## Test Impact
Applied yaml to running cluster (nightly odh)

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.

Screenshot 2024-06-11 at 9 49 30 AM
Screenshot 2024-06-11 at 9 50 09 AM
27-8b9d-200427768137)
Screenshot 2024-06-11 at 9 49 39 AM

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (manaswinidas) [inline manifests/overlays/apps/base/elastic/elastic-app.yaml:51]: Can we provide the full form of ELSER in the first usage?

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (manaswinidas) [inline manifests/overlays/apps/base/elastic/elastic-app.yaml:59]: Elastic Search is written in two ways... `Elastic Search` and `Elasticsearch`. Is this intentional?

Comment (dgutride) [reply manifests/overlays/apps/base/elastic/elastic-app.yaml:59]: Yes  - this was provided by the vendor.  I've asked Kezia to review this on the cluster with them.

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (dgutride) [reply manifests/overlays/apps/base/elastic/elastic-app.yaml:51]: This was provided by the vendor - so I would not change any of the wording unless there are spelling issues.

**Comment by dgutride** [comment]
This is also being reviewed by PM and CCS - please hold on merging until we have their approval.

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (andrewballantyne) [inline manifests/overlays/apps/base/elastic/elastic-app.yaml:1]: I think you want this for RHOAI not everything... right?

We technically have 3 buckets
* ODH
* RHOAI onprem
* RHOAI managed

I feel this is for both RHOAI instances, but maybe just managed?

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (lucferbux) [inline manifests/overlays/apps/base/elastic/elastic-app.yaml:43]: @dgutride Just a question here, the logic we have in place is using `startsWith`, just in case we have multiple csv that have the same prefix. I don't know wether we wanna add support from the Openshift ECK operator `elasticsearch-operator` too, in that case we might wanna change the `csvName`

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (dgutride) [reply manifests/overlays/apps/base/elastic/elastic-app.yaml]: The name is on this line in their repo:  https://github.com/elastic/cloud-on-k8s/blob/8270ad0ae9aebbc39a31fede2c2eb90def222d5b/hack/operatorhub/config.yaml#L42

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (lucferbux) [reply manifests/overlays/apps/base/elastic/elastic-app.yaml:12]: Ok, I've created https://issues.redhat.com/browse/RHOAIENG-8556 to track that effort

File: manifests/overlays/apps/base/elastic/elastic-app.yaml
Comment (lucferbux) [reply manifests/overlays/apps/base/elastic/elastic-app.yaml:1]: Yes, it seems that it's for both environments

**Review by lucferbux — APPROVED** [review]
/lgtm

Tested in cluster, added the operator and it appears in the enabled page.
Screenshot 2024-06-14 at 12 43 12
Screenshot 2024-06-14 at 12 38 09

**Pull request merged**
